compiler/rustc_hir/src/hir.rs

```diff
@@ -401,6 +401,7 @@ pub enum GenericBound<'hir> {
     Trait(PolyTraitRef<'hir>, TraitBoundModifier),
     // FIXME(davidtwco): Introduce `PolyTraitRef::LangItem`
     LangItemTrait(LangItem, Span, HirId, &'hir GenericArgs<'hir>),
+    Unsized(Span),
```

Comment:
Hm, I guess leaving off the `HirId` completely is one way to work around breaking the HIR invariant. Will the lack of `HirId` here cause problems later on, though?

Reply:
I'm really not sure. It _seems_ to me that it shouldn't be a problem, but could envision a future refactor causing issues with this approach. This variant is only used for signaling at the moment (we used to just drop this case on the floor early, as seen on `compiler/rustc_ast_lowering/src/item.rs`).